File: integrations/scala/src/main/scala/angular/ts/Geolocation.scala
package angular.ts

import scala.scalajs.js

/** Plain location snapshot written by the AngularTS `geolocation` directive. */
@js.native
trait GeolocationValue extends js.Object:
  val latitude: Double = js.native
  val longitude: Double = js.native
  val accuracy: Double = js.native
  val altitude: Double | Null = js.native
  val altitudeAccuracy: Double | Null = js.native
  val heading: Double | Null = js.native
  val speed: Double | Null = js.native
  val timestamp: Double = js.native

File: integrations/wasm/rust/crates/angular-ts/tests/geolocation.rs
use angular_ts::GeolocationValue;

#[test]
fn geolocation_value_preserves_nullable_coordinates() {
    let value = GeolocationValue {
        latitude: 56.9496,
        longitude: 24.1052,
        accuracy: 4.5,
        altitude: None,
        altitude_accuracy: Some(8.0),
        heading: Some(90.0),
        speed: None,
        timestamp: 1_788_192_000_000.0,
    };

    assert_eq!(value.latitude, 56.9496);
    assert_eq!(value.altitude, None);
    assert_eq!(value.altitude_accuracy, Some(8.0));
}
